rootio/tchain.go

```diff
@@ -4,8 +4,94 @@
 
 package rootio
 
+import (
+	"fmt"
+	"reflect"
+)
+
 type tchain struct {
-	trees []Tree
+	//	trees []Tree
+	trees []itree
+	cur   int
+	tree  *itree
+}
+
+type itree struct {
+	tree   Tree
+	offset int64 // number of entries before the current one
+	total  int64 // total number of entries iterated over
+}
+
+func (t itree) Entries() int64 {
+	return t.tree.Entries()
+}
+
+func (t *itree) loadEntry(i int64) error {
+	j := i - t.offset
+	return t.tree.loadEntry(j)
+}
+
+type ibranch struct {
+	br     Branch
+	offset int64 // number of entries before the current one
+	total  int64 // total number of entries iterated over
+}
+
+func (b *ibranch) loadEntry(i int64) error {
+	j := i - b.offset
+	return b.br.loadEntry(j)
+}
+
+func (b *ibranch) Branches() []Branch {
+	return b.br.Branches()
+}
+
+func (b *ibranch) Leaves() []Leaf {
+	return b.br.Leaves()
+}
+
+func (b *ibranch) Leaf(name string) Leaf {
+	return b.br.Leaf(name)
+}
+
+func (b *ibranch) setTree(t Tree) {
+	b.br.setTree(t)
+}
+
+func (b *ibranch) getTree() Tree {
+	return b.br.getTree()
+}
+
+func (b *ibranch) getReadEntry() int64 {
+	return b.br.getReadEntry()
+}
+
+func (b *ibranch) getEntry(i int64) {
+	b.br.getEntry(i)
```

Comment:
here the `i` should also be offset'ed by `b.offset` as we do in `ibranch.loadEntry`